### ISSUE TITLE
Compiler bug: Decorator on class where a member method calls a static method

### ISSUE THREAD

**Issue by josundt** [Microsoft/TypeScript#16680]
**TypeScript Version:**  2.4.0

**Code**

```ts
export function testDecorator(data: string): (classType: Function) => void {
    return (classType: Function): void => {
        classType.prototype.__test_decorator_data__ = data;
    };
}

@testDecorator("hello")
export class TestClass {
    doSomething(): void {
        TestClass.doSomethingStatic();
    }

    static doSomethingStatic(): void {
        console.log("Static method called.");
    }
}
```

CompilerOptions:
```json
{
  "compilerOptions": {
    "target": "es5",
    "strict": true,
    "experimentalDecorators": true
  }
}
```

**Expected behavior:**
Compiled JavaScript should not declare variable after return statement.

**Actual behavior:**
***Runtime problems were detected in our project after minifying compiled JavaScript with uglify2 and running it in browsers Edge (ChakraCore) and Internet Explorer 11.*** Chrome and Firefox did not have this problem.
Below is the actual JavaScript compilation output and you will find the variable that has been declared after a return statement. 

Output:
```JavaScript
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
function testDecorator(data) {
    return function (classType) {
        classType.prototype.__test_decorator_data__ = data;
    };
}
exports.testDecorator = testDecorator;
var TestClass = (function () {
    var TestClass = TestClass_1 = function TestClass() {
    };
    TestClass.prototype.doSomething = function () {
        TestClass_1.doSomethingStatic();
    };
    TestClass.doSomethingStatic = function () {
        console.log("Static method called.");
    };
    TestClass = TestClass_1 = __decorate([
        testDecorator("hello")
    ], TestClass);
    return TestClass; // (Editors note): Return statement...
    var TestClass_1; // (Editors note): ...followed by a variable declaration
}());
exports.TestClass = TestClass;
```

**Comment by Kovensky** [comment]
That code looks correct, as does the result when processed by uglify. Seems like this is running into an engine bug?

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
